src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1316,15 +1316,30 @@ private void computeChangeLog(GitClient git, Revision revToBuild, TaskListener l
             boolean exclusion = false;
             ChangelogToBranch changelogToBranch = getExtensions().get(ChangelogToBranch.class);
             if (changelogToBranch != null) {
-                listener.getLogger().println("Using 'Changelog to branch' strategy.");
                 // JENKINS-27080 NFR to allow to use environment variables in "Changelog to branch" behavior
                 String remote = getParameterString(changelogToBranch.getOptions().getCompareRemote(), env);
                 String target = getParameterString(changelogToBranch.getOptions().getCompareTarget(), env);
-                // JENKINS-51633 Improvement to allow leaving empty remote in order to compare to a local file
-                String ref = remote == null || "".equals(remote) ? target : remote +"/"+target;
-                changelog.excludes(ref);
+                // JENKINS-51633 Improvement to allow leaving empty remote in order to compare to a local reference
+                String ref = changelogToBranch.getOptions().isLocalTarget() ? target : remote +"/"+target;
                 exclusion = true;
-            } else {
+		if ( changelogToBranch.getOptions().isLocalTarget() ) {
+		    String localBranchName = env.get(GIT_LOCAL_BRANCH);
+
+		    if (localBranchName == null) {
+			String remoteBranchName = env.get(GIT_BRANCH);
+			exclusion=false;
+			// treating as if the option had not been specified at all, but warning the user
+			listener.getLogger().println("Ignoring 'Changelog to branch' strategy to exclude: "+ref
+						     +" because there is no local branch name for remote branch "+remoteBranchName
+						     +".  Did you mean to set remote or add Check out to specific local branch option?");
+		    }
```

Comment:
I think this should just set the "extension requires workspace polling" field when the remote is non empty, since this will force us to poll from within the workspace in this case.